src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -1621,6 +1522,31 @@ void RequireDynamicallyAccessedMembers (ref ReflectionPatternContext reflectionC
 			reflectionContext.RecordHandledPattern ();
 		}
 
+		static string[] GetDidNotMatchRequirementsWarningMessageArgs (IMetadataTokenProvider source, IMetadataTokenProvider target)
+		{
+			List<string> args = new List<string> ();
```

Comment:
I would prefer to keep the code spread around above. It will make it easier to customize the message if we wanted to. This way the above switch could really just determine the warning code and the warning and message construction could be unified. I know it goes a bit against code deduplication, but the point here is to make it easy to improve UX going forward.